tests/test_shadow_fork.py

```
@@ -76,28 +76,48 @@ def test_create_fails_if_base_branch_missing(self, tmp_path):
-            ]
-
-            with pytest.raises(RuntimeError, match="does not exist"):
-                fork.create(base_branch="nonexistent")
+        # Mock side effect that handles fallback logic
+        # 1. Get current branch (HEAD)
+        # 2. Check uncommitted changes
+        # 3. Verify base branch (fails for nonexistent)
+        # 4. Verify fallback to HEAD (succeeds)
+        # 5. Create shadow branch
+        # 6. Install push guard
+        def mock_git_side_effect(*args, **kwargs):
```

Ruff (0.15.2)

[warning] Missing return type annotation for private function `mock_git_side_effect`

(ANN202)